isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -1864,6 +1860,14 @@ namespace Isis {
       swap(copy);
     }
 
+    m_vertexMap.clear();
+    VertexIterator v, vend;
+    for (boost::tie(v, vend) = vertices(m_controlGraph); v != vend; ++v) {
+      ImageVertex imVertex = *v;
+      QString serialNum = m_controlGraph[*v].serial;
+      m_vertexMap[serialNum] = imVertex;
+    }
+
```

Comment:
I'm worried that maybe this needs to be in swap(). If you just call swap on a ControlNet right now, the vertexmap will be completely broken. In fact I think the vertex maps in both ControlNets will be broken.

Reply:
:+1: I'll go ahead and move this in there and make sure everything still works. I don't remember why I moved this out of ControlNet::swap() in the first place.

Reply:
I think you'll need to go through this process for both cubes that are being swapped, self and other.

Reply:
Good point, thank you.